### PR TITLE
IPaddr2: fix send_arp issues and support arping command

### PR DESCRIPTION
**Pull request by kskmori**
This series of patches is to fix a bunch of issues related to send_arp so far (__Known issues:__ below) as well as to support `arping` command instead of send_arp.
This pull request should not change any behavior for the existing configuration (except for bug fixes).

I would appreciate your comments and feedback, particularly from Debian user (which has two versions of arping and I don't know how they are used), and from Cluster IP users (which I haven't used so much).

Regards,

### Known issues: ###
* need an option for  sending ARP REPLY (`arping -A`) 
  -  some switches does not accept ARP REQUEST(arping -U, send_arp) but do only ARP REPLY (arping -A)
* send_arp: arp_interval does not work
  - http://lists.clusterlabs.org/pipermail/users/2016-October/004180.html
* send_arp: src_hw_addr, broadcast_ip_addr, and netmask are not set properly which causes clusterip unavailable
  - ​https://github.com/ClusterLabs/resource-agents/issues/860
* send_arp: improper documentation
  - https://github.com/ClusterLabs/resource-agents/issues/860
* IPaddr2: arp_mac is no longer in use but still described in the metadata
  - removed with the commit 2852f51 and 0295fa1
* an output from send_arp is not logged when run in a background
  - https://github.com/ClusterLabs/resource-agents/pull/958

### Proposals to fix: ###
* extend `arp_sender` parameter to allow to choose the standard arping bundled in the distro. 
  - aims to replace send_arp binary with arping as default in the future
  - also considering a variant of arping command (iputils_arping, libnet_arping)
* add a new `send_arp_opts` parameter 
  - allow to send ARP REPLY instead of ARP REQUEST
* send_arp.{libnet, linux} are left only for the backward compatibility with minimal changes
  - basically fixes documentation errors only
  - the usage of send_arp.linux shows options that are currently used as a drop-in replacement of send_arp.libnet
* let `arp_interval` parameter as deprecated and document it
  - it does not seem to be used frequently and only limited tools support it
* do not send gratuitous ARPs in the Cluster IP configuration
  - it does not seem to be necessary because the cluster MAC will not change and nothing needs to be updated on the peers (point me if it's wrong)
  - sending a physical MAC does cause a problem 
* use ocf_log for the output from send_arp command

### Related duscussion: ###
* The great summary of the history by lge
  - ​https://github.com/ClusterLabs/resource-agents/issues/860#issuecomment-263335126
* suggestion of replacing send_arp with arping in the past
  - https://lists.gt.net/linuxha/pacemaker/58110#58110
  - https://github.com/ClusterLabs/resource-agents/issues/860#issuecomment-263335126
  - https://github.com/ClusterLabs/resource-agents/pull/654#issuecomment-128015953
* a different approach to fix send_arp usage
  - https://github.com/ClusterLabs/resource-agents/pull/961

### CODE REVIEW

File: heartbeat/IPaddr2 [hunk 1]
```diff
@@ -701,6 +703,23 @@ log_arp_sender() {
...
+    local cmdline
+    local pid
+    local rc
+
+    cmdline="$@"
+
+    $cmdline &
+    pid=$!
+    echo "$pid" > $SENDARPPIDFILE
+    wait $pid
```
Comment (dmuhamedagic) [inline heartbeat/IPaddr2]: I'm a bit confused here. What's the purpose of running the program in the background when immediately we wait for it?

Comment (kskmori) [inline heartbeat/IPaddr2]: This is to create and remove the pid file of arping as similar manner  as the send_arp binary does.
Running it in the background once is to store the pid of arping via $! to the pid file (which could be used later to clean up if it's still running when arp_bg=true), and waiting it here is to remove the pid file as soon as arping exits at the next line.

**Comment by kskmori** [comment]
Rebased to the master so that it could be merged easily.
I'm going to merge this until  4.1.0-rc1 if there is no objection.

I'm wondering why Travis-CI fails and now looking into it. There is no issue here, neither before rebasing.

**Comment by krig** [comment]
Hi, the CI failure is unrelated, currently all CI is failing due to a problem with the Travis base installation.

**Comment by kskmori** [comment]
Oh, I see. Thanks!

**Pull request merged**